src/k8s/api/v1/types.go

```diff
@@ -119,20 +119,74 @@ func (c ClusterStatus) String() string {
 	} else {
 		result.WriteString("no")
 	}
-	result.WriteString("\n\n")
-	result.WriteString("control-plane nodes:\n")
-	for _, member := range c.Members {
-		// There is not much that we can do if the hostport is wrong.
-		// Thus, ignore the error and just display an empty IP field.
-		apiServerIp, _, _ := net.SplitHostPort(member.Address)
-		result.WriteString(fmt.Sprintf("  %s: %s\n", member.Name, apiServerIp))
+	result.WriteString("\n")
+	result.WriteString("datastore:\n")
+
+	voters := make([]NodeStatus, 0, len(c.Members))
+	standBys := make([]NodeStatus, 0, len(c.Members))
+	spares := make([]NodeStatus, 0, len(c.Members))
+	for _, node := range c.Members {
+		switch node.DatastoreRole {
+		case DatastoreRoleVoter:
+			voters = append(voters, node)
+		case DatastoreRoleStandBy:
+			standBys = append(standBys, node)
+		case DatastoreRoleSpare:
+			spares = append(spares, node)
+		}
+	}
+	if len(voters) > 0 {
+		result.WriteString(fmt.Sprintf("  master-nodes:\n"))
+		for _, voter := range voters {
+			result.WriteString(fmt.Sprintf("    - %s\n", voter.Address))
+		}
+	} else {
+		result.WriteString(fmt.Sprintf("  master-nodes: none\n"))
+	}
+	if len(standBys) > 0 {
+		result.WriteString(fmt.Sprintf("  standy-nodes:\n"))
+		for _, standBy := range standBys {
+			result.WriteString(fmt.Sprintf("    - %s\n", standBy.Address))
+		}
+	} else {
+		result.WriteString(fmt.Sprintf("  standy-nodes: none\n"))
+	}
+	if len(spares) > 0 {
+		result.WriteString(fmt.Sprintf("  spare-nodes:\n"))
+		for _, spare := range spares {
+			result.WriteString(fmt.Sprintf("    - %s\n", spare.Address))
+		}
+	} else {
+		result.WriteString(fmt.Sprintf("  spare-nodes: none\n"))
```

Comment:
Could a function be added to reduce repetitiveness of building the string?

Reply:
Hm, I think it may be slightly more readable but not much.

Also, I would put such function into the `utils` package but we cannot do this as this would cause cyclic dependencies.

A bit of repetitiveness is the price we pay with the explicit configuration, I guess